great_expectations/profile/user_configurable_profiler.py

```diff
@@ -0,0 +1,1122 @@
+                "ignored_columns": A list of columns for which you would like to NOT create expectations
+                "not_null_only": Boolean, default False. By default, each column is evaluated for nullity. If the column
+                    values contain fewer than 50% null values, then the profiler will add
+                    `expect_column_values_to_not_be_null`; if greater than 50% it will add
+                    `expect_column_values_to_be_null`. If not_null_only is set to True, the profiler will add a
+                    not_null expectation irrespective of the percent nullity (and therefore will not add an
+                    `expect_column_values_to_be_null`
+                "primary_or_compound_key": A list containing one or more columns which are a dataset's primary or
+                    compound key. This will create an `expect_column_values_to_be_unique` or
+                    `expect_compound_columns_to_be_unique` expectation. This will occur even if one or more of the
+                    primary_or_compound_key columns are specified in ignored_columns
+                "semantic_types": A dictionary where the keys are available semantic_types (see ProfilerSemanticTypes)
+                    and the values are lists of columns for which you would like to create semantic_type specific
+                    expectations e.g.:
+                    "semantic_types": { "value_set": ["state","country"], "numeric":["age", "amount_due"]}
+                "table_expectations_only": Boolean, default False. If True, this will only create the two table level expectations
+                    available to this profiler (`expect_table_columns_to_match_ordered_list` and
+                    `expect_table_row_count_to_be_between`). If a primary_or_compound key is specified, it will create
+                    a uniqueness expectation for that column as well
+                "value_set_threshold": Takes a string from the following ordered list - "none", "one", "two",
+                    "very_few", "few", "many", "very_many", "unique". When the profiler runs withou a semantic_types
+                    dict, each column is profiled for cardinality. This threshold determines the greatest cardinality
+                    for which to add `expect_column_values_to_be_in_set`. For example, if value_set_threshold is set to
+                    "unique", it will add a value_set expectation for every included column. If set to "few", it will
+                    add a value_set expectation for columns whose cardinality is one of "one", "two", "very_few" or
+                    "few". The default value is "many". For the purposes of comparing whether two tables are identical,
+                    it might make the most sense to set this to "unique"
+        """
+        self.dataset = dataset
+        self.column_info = {}
+
+        self.semantic_type_dict = semantic_types_dict
```

Comment:
Unless there is a reason, I would suggest standardizing on one name here to avoid confusion (`type` vs `types`)